gto-support-picasso/src/test/java/org/ccci/gto/android/common/picasso/RequestCreatorTest.kt

```diff
@@ -119,6 +121,30 @@ class RequestCreatorTest {
         verify(picasso).cancelRequest(target)
     }
 
+    @Test
+    fun `getBitmap() should cancel Picasso request on main thread`() {
+        val thread = HandlerThread("").apply { start() }
+        assertTrue(thread.isAlive)
+        Dispatchers.setMain(Handler(thread.looper).asCoroutineDispatcher())
+        var cancelRequestThread: Thread? = null
+        val picasso = mock<Picasso> {
+            on { cancelRequest(any<Target>()) } doAnswer { cancelRequestThread = Thread.currentThread() }
+        }
+
+        val request = TargetCapturingRequestCreator(picasso)
+        val target = runBlocking {
+            val task = launch(TestCoroutineDispatcher()) { request.getBitmap() }
+            val target = request.targets.receive().get()!!
+            task.cancel()
+            return@runBlocking target
+        }
+        thread.quitSafely()
+        thread.join()
```

Comment:
this causes the Android `HandlerThread` to process any pending tasks on its queue that haven't executed yet.